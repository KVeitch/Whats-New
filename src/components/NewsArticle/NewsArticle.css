article {
  width: calc(50% - 25px);
  min-width: 450px;
  color:black;
  display:inline-block;
  margin: 10px 10px;
  background-color:white;
  height: 500px;
  border-radius: 5px;
  background-color:rgba(255, 255, 255, 0.9);
  cursor: default;
}

article:hover {
  box-shadow: 5px 5px #121e3f;
}

.article-image {
  width: 100%;
}

.image-container {
  width: 100%;
  height: 200px;
  border-radius: 5px 5px 0 0;
  overflow: hidden;
}
article h2{
  padding: 0 5px;
  height:100px;
  margin-top:10px;
}

article p {
  font-size: 18px;
  padding: 0 5px;
  height: 140px;
}

article a, footer {
  background-color: #8c7ae6;
  font-size:24px;
  color: white;
  text-decoration: none;
  /* height:50px; */
  border-radius: 0 0 5px 5px;
}

footer {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 20px;
}

 footer img {
   height:50px;
 }

