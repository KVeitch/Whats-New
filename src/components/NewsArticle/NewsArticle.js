import React from 'react';
import './NewsArticle.css';

const NewsArticle = (story) => {
  return <article>
    <div className='image-container'>
      <img className='article-image' src={story.img} alt='Story related graphic'/>
    </div>
    <h2>{story.headline}</h2>
    <p>{story.description}</p>
    <a href={story.url} target="blank">
      <footer>Link To Article</footer>
    </a>
  </article>
 }

export default NewsArticle;

