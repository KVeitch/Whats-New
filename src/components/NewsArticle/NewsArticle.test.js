import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle';


describe('NewsArticle', () =>{
  it('should match the snapshot', () => {
    const newsStory = {
      headline:'Big News Here',
      description:'News Descriptiopn here',
      url:'http://www.moretesting.com'
    }
    const wrapper = shallow(<NewsArticle
      story={newsStory}
    />);

    expect(wrapper).toMatchSnapshot()
  });

})