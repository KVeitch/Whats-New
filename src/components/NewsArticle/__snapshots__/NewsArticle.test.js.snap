// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the snapshot 1`] = `
<article>
  <div
    className="image-container"
  >
    <img
      alt="Story related graphic"
      className="article-image"
    />
  </div>
  <h2 />
  <p />
  <a
    target="blank"
  >
    <footer>
      Link To Article 
      <img
        alt="arrow pointing to the right"
        src="arrow-to-right.svg"
      />
    </footer>
  </a>
</article>
`;
